src/components/modals/ManageAntModal/ManageAntModal.tsx

```diff
@@ -0,0 +1,236 @@
+  const isMobile = useIsMobile();
+  // todo: manage asset modal writes asset modifications to contract. It will auto detect if the asset is an ANT, name, or undername.
+  // if the asset is a name, it will write modifications to the registry. If its an undername, it will write mods to the ant. If its an ant, it will write mods to the ant.
+
+  useEffect(() => {
+    setAssociatedNames(getAssociatedNames());
+  }, [contractId, state]);
+
+  function handleClickOutside(e: any) {
+    if (modalRef.current && modalRef.current === e.target) {
+      setShowModal(false);
+    }
+    return;
+  }
+
+  function getAssociatedNames() {
+    const domains: string[] = [];
+    Object.entries(arnsSourceContract.records).map(([name, id]) => {
+      if (id === contractId) {
+        domains.push(name);
+      }
+    });
+    return domains;
+  }
+
+  return (
+    // eslint-disable-next-line
+    <div
+      className="modal-container"
+      style={{ background: '#1E1E1E' }}
+      ref={modalRef}
+      onClick={handleClickOutside}
+    >
+      <div className="flex-column" style={{ margin: '10%' }}>
+        <div className="flex flex-row">
+          <span
+            className="flex section-header"
+            style={{
+              justifyContent: 'flex-start',
+              width: '100%',
+            }}
+          >
+            <NotebookIcon width={25} height={25} fill={'var(--text-white)'} />
+            &nbsp;Manage ANT:&nbsp;
+            <span>
+              {isMobile ? (
+                <CopyTextButton
+                  displayText={`${contractId.slice(0, 10)}...${contractId.slice(
+                    -10,
+                  )}`}
+                  copyText={contractId}
+                  size={24}
+                  wrapperStyle={{
+                    display: 'flex',
+                    justifyContent: 'center',
+                    alignItems: 'center',
+                    textColor: 'var(--bright-white)',
+                  }}
+                />
+              ) : (
+                <CopyTextButton
+                  displayText={contractId}
+                  copyText={contractId}
+                  size={24}
+                  wrapperStyle={{
+                    display: 'flex',
+                    justifyContent: 'center',
+                    alignItems: 'center',
+                    textColor: 'var(--bright-white)',
+                  }}
+                />
+              )}
+            </span>
+          </span>
+        </div>
+        <table className="assets-table">
+          <RowItem
+            details={[
+              'Status:',
+              <TransactionStatus
+                key={`${contractId}-confirmations`}
+                confirmations={confirmations}
+              />,
+            ]}
+            bgColor={'#1E1E1E'}
+            textColor={'var(--text-white)'}
+          />
+          <RowItem
+            details={[
+              `Associated Names: ${associatedNames.length}`,
```

Comment:
nit: the design does not include the count. if we are going to include it I would put it as

`Associated Names (COUNT): `,

<img width="548" alt="image" src="https://user-images.githubusercontent.com/12800001/214073879-30ccfd72-630a-418f-93ca-982c22ecb7b0.png">

Reply:
Right - need to discuss this at the design meet tomorrow. An ant can have many names associated with it, and they can take up more space than is available - so I added `count` to display the total number. Still waiting on alot of my questions to be answered, but IK lucas has been working on it today so hopefully ill get one soon.